product-service/services/product.service.ts

```diff
@@ -1,107 +1,109 @@
-            count: 200
-        },
-        {
-            id: '4-Pound-Toothed-Bush-Hammer',
-            title: '4-Pound Steel Toothed Bush Hammer',
-            description: 'Perfect for shaping, texturizing, and smoothing soft stone, such as marble or limestone.',
-            weight: 1800,
-            img: 'https://images-na.ssl-images-amazon.com/images/I/61PSkLLka-L._AC_SL1500_.jpg',
-            creationDate: new Date ('2020-10-11'),
-            price: 87,
-            count: 10
-        },
-        {
-            id: 'Round-Mallet-Bell-Hammer',
-            title: 'Round Mallet / Bell Hammer',
-            description: 'Our High Quality Round Hand Hammer is hardened for more power and better wear. The round striking surface makes it easier to deliver a solid blow every time. ',
-            weight: 600,
-            img: 'https://images-na.ssl-images-amazon.com/images/I/51hdPZCJC3L._AC_SL1000_.jpg',
-            creationDate: new Date ('2020-01-09'),
-            price: 58,
-            count: 13
-        }
-    ];
-  
-  
     async getList() : Promise<ProductModel[]>  {
-        return Promise.resolve(this.products);    
+        const client = new Client(dbOptions);
+        await client.connect();
+    
+        try {
+            const result =  await client.query(GET_PRODUCTS_SQL);
+            return result.rows;
+            
+        } finally {
+            if (client) {
+                client.end();
+            }
+        }
     }
 
     async getProductById(id: string) : Promise<ProductModel> {
-        return this.getList()
-            .then((products) => { 
-                return products.find(product => product.id === id)
-            });
+        const client = new Client(dbOptions);
+        await client.connect();
+    
+        try {
+            const result =  await client.query(GET_PRODUCTS_BY_ID_SQL, [id]);
+            return result.rows[0];
+            
+        } finally {
+            if (client) {
+                client.end();
+            }
+        }
     }
   
-    // createProduct(item: ProductModel) {
-    //     return this.products.push(item);
-    // }
+    async createProduct(item: ProductModel) : Promise<ProductModel> {
+        const client = new Client(dbOptions);
+        await client.connect();
+    
+        try {
+            await client.query('BEGIN');
+
+            // url_id, title, description, weight, img, creation_date, price
+            await client.query(CREATE_PRODUCT_SQL, 
+                    [item.id, 
+                    item.title, 
+                    item.description, 
+                    item.weight,
+                    item.img,
+                    item.creationDate,
+                    item.price]);
+
+            await client.query(CREATE_STOCK_SQL, [item.id, item.count]);
+
+            await client.query('COMMIT');
+
+            // return new created product
+            const result =  await client.query(GET_PRODUCTS_BY_ID_SQL, [item.id]);
```

Comment:
to avoid another DB call you could add `RETURNING` to your query

Reply:
Thank you for help!
I see, it is very useful feature of postgres!